account-kit/smart-contracts/src/ma-v2/account/nativeSMASigner.ts

```diff
@@ -108,8 +108,11 @@ export const nativeSMASigner = (
     ): Promise<Hex> => {
       // the accounts domain already gives replay protection across accounts for deferred actions, so we don't need to apply another wrapping
       const isDeferredAction =
-        typedDataDefinition?.primaryType === "DeferredAction" &&
-        typedDataDefinition?.domain?.verifyingContract === accountAddress;
+        typedDataDefinition.primaryType === "DeferredAction" &&
+        typedDataDefinition.domain != null &&
+        // @ts-expect-error the domain type I think changed in viem, so this is not working correctly
```

Comment:
❓ does this need to be addressed before merging? or is it "working" fine, just the types are wrong?

Reply:
it's working fine, I just think the types have changed

Reply:
ok great